App/Views/Pages/TrayWindowMainPage.xaml

```diff
@@ -118,22 +115,50 @@
                                 HorizontalAlignment="Stretch"
                                 Spacing="10">
 
+                                <StackPanel.Resources>
+                                    <converters:StringToBrushSelector
+                                        x:Key="StatusColor"
+                                        SelectedKey="{x:Bind Path=ConnectionStatus, Mode=OneWay}">
+
+                                        <converters:StringToBrushSelectorItem>
+                                            <converters:StringToBrushSelectorItem.Value>
+                                                <SolidColorBrush Color="#8e8e93" />
+                                            </converters:StringToBrushSelectorItem.Value>
+                                        </converters:StringToBrushSelectorItem>
+                                        <converters:StringToBrushSelectorItem Key="Red">
+                                            <converters:StringToBrushSelectorItem.Value>
+                                                <SolidColorBrush Color="#ff3b30" />
+                                            </converters:StringToBrushSelectorItem.Value>
+                                        </converters:StringToBrushSelectorItem>
+                                        <converters:StringToBrushSelectorItem Key="Yellow">
+                                            <converters:StringToBrushSelectorItem.Value>
+                                                <SolidColorBrush Color="#ffcc01" />
+                                            </converters:StringToBrushSelectorItem.Value>
+                                        </converters:StringToBrushSelectorItem>
+                                        <converters:StringToBrushSelectorItem Key="Green">
+                                            <converters:StringToBrushSelectorItem.Value>
+                                                <SolidColorBrush Color="#34c759" />
+                                            </converters:StringToBrushSelectorItem.Value>
+                                        </converters:StringToBrushSelectorItem>
+                                    </converters:StringToBrushSelector>
+                                </StackPanel.Resources>
```

Comment:
I find this way less readable than the `AgentStatusToColorConverter`

Reply:
The reason I added `DependencyObjectSelector` was for being able to convert from an enum to a `ThemeResource` color, which is necessary in `FileSyncListMainPage` because we're mapping to WinUI resource colors that change when the system theme changes.

In the case of `AgentViewModel.ConnectionStatus`, we aren't using theme resources and instead using hardcoded colors which look OK on both light and dark themes (and probably not high contrast themes).

Since we're using hardcoded colors we could keep the converter, but it seems weird to use one method in one place and another in a different place. I agree it's less readable, though.

Also, if we decide to use theme resources for connection status (which we probably should TBH) we will need to switch to this method anyways

Reply:
Is there some way we could instantiate this resource in plain C# ? I feel like not paying the [Angle Bracket Tax](https://blog.codinghorror.com/xml-the-angle-bracket-tax/) alone would help a lot.

Reply:
Discussed on Slack, it doesn't seem to be possible or I cannot figure out how to do it. :/